### PR TITLE
Feature/atom progress bar

### PR DESCRIPTION
**Pull request by juanmaguitar**
https://jira.scmspain.com/browse/SUIC-32
Demo online can be checked at → https://sui-components-feature-atom-progress-bar.now.sh/workbench/atom/progressBar/demo

### CODE REVIEW

File: components/atom/progressBar/src/index.js [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+        </div>
+        {indicatorBottom && this.renderIndicator(indicatorBottom)}
+      </div>
+    )
+  }
+}
+
+AtomProgressBar.displayName = 'AtomProgressBar'
+
+// Remove these comments if you need
```
Comment (midudev) [inline components/atom/progressBar/src/index.js]: Remove the comments not needed. :)

File: components/atom/progressBar/src/index.js [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+import PropTypes from 'prop-types'
+import cx from 'classnames'
+
+const BASE_CLASS = 'sui-AtomProgressBar'
+const CLASS_INDICATOR = 'sui-AtomProgressBar-indicator'
+const CLASS_CONTAINER_BAR = 'sui-AtomProgressBar-container'
+const CLASS_BAR = 'sui-AtomProgressBar-bar'
+
+class AtomProgressBar extends PureComponent {
+  renderIndicator(indicatorBottom) {
```
Comment (midudev) [inline components/atom/progressBar/src/index.js]: Usually, I would like more a component itself than a render method.

Comment (midudev) [reply components/atom/progressBar/src/index.js]: Also, IMHO it's a bit weird the implementation of this method. As the indicatorBottom is retrieved from param but still is accessing from props the rest of props. I would or use only named params or only use props.

File: components/atom/progressBar/src/index.scss [hunk 3]
```diff
@@ -0,0 +1,42 @@
+@import '~@schibstedspain/sui-theme/lib/index';
+
+$bg-progress-bar: $c-gray-lightest;
```
Comment (midudev) [inline components/atom/progressBar/src/index.scss]: Should these be default values in order to be custom or are not customizable any of them?

File: components/atom/progressBar/src/index.scss [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+    &--bottom {
+      margin-top: $m-base;
+    }
+    
+  }
+
+  &-container {
+    background: $bg-progress-bar;
+    border: 1px solid $bc-progress-bar;
+    border-radius: 25px;
```
Comment (midudev) [inline components/atom/progressBar/src/index.scss]: As the 25px is used twice, might be is a good idea to extract to a variable in order to change once and get the value. Even if not customizable.

**Review by midudev — DISMISSED** [review]
I rather like to see the usage of the progress HTML element but I assume it has some styling considerations that could make it harder to work with.

File: components/atom/progressBar/src/index.js (hunk 2 above)
Comment (juanmaguitar) [reply components/atom/progressBar/src/index.js]: 👍

File: components/atom/progressBar/src/index.scss (hunk 4 above)
Comment (juanmaguitar) [reply components/atom/progressBar/src/index.scss]: 👍

**Comment by juanmaguitar** [comment]
Regarding this → https://github.com/SUI-Components/sui-components/pull/542#pullrequestreview-163751283
Yes, with the `progress` tag the styling is VERY limited and every browser will display its own version.

File: components/atom/progressBar/src/index.scss (hunk 3 above)
Comment (juanmaguitar) [reply components/atom/progressBar/src/index.scss]: I don't see anything in the specification regarding this but I understand that these colors should be customizable

Comment (midudev) [reply components/atom/progressBar/src/index.scss]: Ok! We could ask UX guys in order to certify this, but if the colors are customizable then we should use `!default` in these variables.

**Review by midudev — DISMISSED** [review]

**Review by carlosvillu — DISMISSED** [review]

**Review by midudev — DISMISSED** [review]

**Review by ddmarin94 — DISMISSED** [review]

**Review by devictoribero — DISMISSED** [review]

File: components/atom/progressBar/src/index.scss [hunk 5]
```diff
@@ -0,0 +1,49 @@
+@import '~@schibstedspain/sui-theme/lib/index';
+
+$bg-progress-bar: $c-gray-lightest !default;
+$bc-progress-bar: $c-gray-light !default;
+$c-progress-bar-fill: $c-primary !default;
+
+$trsdu-progress-bar: 500ms !default;
+$trstf-progress-bar: linear !default;
+
+$br-progress-bar: 25px;
```
Comment (jcger) [inline components/atom/progressBar/src/index.scss]: this was 4px, right?

**Review by ddmarin94 — DISMISSED** [review]

File: components/atom/progressBar/README.md [hunk 6]
```diff
@@ -0,0 +1,38 @@
+# AtomProgressBar
+
+AtomProgressBar is a component that displays a progress bar
```
Comment (jcger) [inline components/atom/progressBar/README.md]: what about `A Progress bar is a visual representation of a specified wait time or the current status of a process.`?

**Review by jcger — DISMISSED** [review]

**Review by ddmarin94 — APPROVED** [review]

**Review by jcger — APPROVED** [review]

**Review by midudev — APPROVED** [review]

File: components/atom/progressBar/src/index.js [hunk 7]
```diff
@@ -0,0 +1,62 @@
...
+import cx from 'classnames'
+
+const BASE_CLASS = 'sui-AtomProgressBar'
+const CLASS_INDICATOR = 'sui-AtomProgressBar-indicator'
+const CLASS_CONTAINER_BAR = 'sui-AtomProgressBar-container'
+const CLASS_BAR = 'sui-AtomProgressBar-bar'
+const CLASS_BAR_ANIMATED = 'sui-AtomProgressBar-bar--animated'
+
+const Indicator = props => {
+  const {indicatorBottom, percentage, indicatorTotal} = props // eslint-disable-line react/prop-types
```
Comment (carlosvillu) [inline components/atom/progressBar/src/index.js:12]: Why are avoid add the prop types definition for this component ?!

**Review by carlosvillu — APPROVED** [review]

**Pull request merged**